core/src/main/java/hudson/util/AtomicFileWriter.java

```diff
@@ -116,7 +143,15 @@ public AtomicFileWriter(@Nonnull Path destinationPath, @Nonnull Charset charset)
             throw new IOException("Failed to create a temporary file in "+ dir,e);
         }
 
-        core = Files.newBufferedWriter(tmpPath, charset);
+        if (LOSE_INTEGRITY) {
+            // We should log it in WARN, but as this code is called often, this would create huge logs in prod system
+            // Hence why a similar log is logged once above in a static block.
+            LOGGER.log(Level.FINE, "WARNING: YOU SET A FLAG THAT COULD LEAD TO DATA LOSS.");
```

Comment:
Seriously, we don't need to warn on every write.  The data loss happens when you're yanking storage/OS out from under the software -- most applications aren't expected to behave in that scenario.

Reply:
We should be issuing a more specific warning, something like "virtualization or storage failures can cause dataloss if this is enabled."  My concern is that in settings where users *do* have reliable storage and are bitten by the IO performance drop they will still be too scared to use this even if we explicitly tell 'em to.